### ISSUE TITLE
False positive SING_SINGLETON_IMPLEMENTS_SERIALIZABLE when class is not a singleton

### ISSUE THREAD

**Issue by colin-janke** [spotbugs/spotbugs#2964]
Code like this (somewhat anonymised, but I think this expresses the idea)

```java
public class MyFancyCollection implements Serializable {
    private static final MyFancyCollection EMPTY = new MyFancyCollection(new IntArraySet(0)) {
        private Object readResolve() {
            return EMPTY;
        }
    };

    private MyFancyCollection(IntSet set) {
        this.set = set;
    }

    public static MyFancyCollection empty() {
        return EMPTY;
    }

    public static MyFancyCollection copyOf(Set<Integer> sourceData) {
        IntSet set = new IntOpenHashSet(sourceData.size());
        sourceData.forEach(i -> set.add(i));
        return new ImmutableCoordinateSet(set);
    }
...
```

is producing SING_SINGLETON_IMPLEMENTS_SERIALIZABLE, when the class is not a singleton. It has a constant  EMPTY field, but that is just utility and there is no real harm to creating multiple empty instances (you will note that the copyOf method does not return EMPTY if the copied set is EMPTY).

**Comment by colin-janke** [comment]
I should add this is with spotbugs v4.8.4

**Comment by JuditKnoll** [comment]
This seems to be the duplicate of https://github.com/spotbugs/spotbugs/issues/2934 (see the second example). It will be fixed by https://github.com/spotbugs/spotbugs/pull/2951.
